Azure data lake input source (#11153)
apache/druid: docs/ingestion/native-batch.md

> You need to include the [`druid-azure-extensions`](../development/extensions-core/azure.md) as an extension to use the Azure input source.

> The Azure input source is used to read objects directly from Azure Blob store or Azure Data Lake sources. Objects can be

Reviewer: ```suggestion
The Azure input source reads objects directly from Azure Blob store or Azure Data Lake sources. You can
```
nit

> specified as a list of file URI strings or prefixes. The Azure input source is splittable and can be used by the [Parallel task](#parallel-task), where each worker task reads a single object.

Reviewer: ```suggestion
specify objects as a list of file URI strings or prefixes. You can split the Azure input source for use with [Parallel task](#parallel-task) indexing and each worker task reads one chunk of the split data.
```
I think we should differentiate between the `single object` and the sections of split out object since we're using `object` as the whole.

|property|description|default|required?|
|--------|-----------|-------|---------|
|type|This should be `azure`.|None|yes|
|uris|JSON array of URIs where the Azure objects to be ingested are located, in the form "azure://\<container>/\<path-to-file\>"|None|`uris` or `prefixes` or `objects` must be set|

> |prefixes|JSON array of URI prefixes for the locations of Azure objects to be ingested, in the form "azure://\<container>/\<prefix\>". Empty objects starting with one of the given prefixes are skipped.|None|`uris` or `prefixes` or `objects` must be set|

Reviewer: ```suggestion
|prefixes|JSON array of URI prefixes for the locations of Azure objects to ingest, in the form "azure://\<container>/\<prefix\>". Empty objects starting with one of the given prefixes are skipped.|None|`uris` or `prefixes` or `objects` must be set|
```
nit

> |objects|JSON array of Azure objects to be ingested.|None|`uris` or `prefixes` or `objects` must be set|

Reviewer: ```suggestion
|objects|JSON array of Azure objects to ingest.|None|`uris` or `prefixes` or `objects` must be set|
```
nit
